pandas/tests/test_categorical.py

```diff
@@ -1227,6 +1227,10 @@ def test_set_categories(self):
         exp_categories = Index(["a", "b", "c", "d"])
         tm.assert_index_equal(cat.categories, exp_categories)
 
+        # all-nan categories GH 18051
+        cat_nan = Categorical([np.nan])
+        assert cat_nan.unique()._codes.flags.writeable
```

Comment:
this are all tests about `set_categories` so it feels a bit strange to put this one in here

Reply:
sure